client/main.js

```diff
@@ -362,35 +365,106 @@ $(document).ready(function() {
   $('#accordion').on('show.bs.collapse', function(e) {
       expandCaret(e.target);
       if ($('a[data-toggle=collapse]').length === $('.fa-caret-down').length) {
-        $('#showAll').text('Collapse all challenges');
-        $('#showAll').addClass('active');
+        mapShowAll.text('Collapse all challenges');
+        mapShowAll.addClass('active');
       }
   }).on('hide.bs.collapse', function(e) {
       collapseCaret(e.target);
       if ($('a[data-toggle=collapse]').length === $('.fa-caret-right').length) {
-        $('#showAll').text('Expand all challenges');
-        $('#showAll').removeClass('active');
+        mapShowAll.text('Expand all challenges');
+        mapShowAll.removeClass('active');
       }
   });
 
-  $('#showAll').on('click', () => {
-    var mapExpanded = $('#showAll').hasClass('active');
+  mapShowAll.on('click', () => {
+    var mapExpanded = mapShowAll.hasClass('active');
     if (!mapExpanded) {
       $.each($('.map-collapse:not(".in")'),
       function(i, div) {
         expandBlock(div);
       });
-      $('#showAll').text('Collapse all challenges');
-      return $('#showAll').addClass('active');
+      mapShowAll.text('Collapse all challenges');
+      return mapShowAll.addClass('active');
     } else {
       $.each($('.map-collapse.in'), function(i, div) {
         collapseBlock(div);
       });
-      $('#showAll').text('Expand all challenges');
-      return $('#showAll').removeClass('active');
+      mapShowAll.text('Expand all challenges');
+      return mapShowAll.removeClass('active');
     }
   });
 
+  // Map live filter
+  mapFilter.on('keyup', () => {
+    if (mapFilter.val().length > 0) {
+      var regex = new RegExp(mapFilter.val().replace(/ /g, '.'), 'i');
+
+      // Hide/unhide challenges that match the regex
+      $('.challenge-title').each((index, title) => {
+        if (regex.test($(title).attr('name'))) {
+          expandBlock($(title).closest('.chapterBlock'));
+          expandBlock($(title).closest('.certBlock'));
+          $(title).removeClass('hidden');
+        } else {
+          $(title).addClass('hidden');
+        }
+      });
+
+      // Hide/unhide blocks with no matches
+      $.each($('.chapterBlock'), function(i, div) {
+        if ($(div).find('.hidden').length ===
+            $(div).find('p').length) {
+                $(div).addClass('hidden');
```

Comment:
Please, use 2 spaces spaces for indentation instead of 4 spaces everywhere here. That brakes the pattern used in the whole file.